dgraph/cmd/graphql/admin/admin.go

```diff
@@ -178,8 +178,37 @@ type adminServer struct {
 	withIntrospection bool
 }
 
-// NewAdminResolver creates a GraphQL request resolver for the /admin endpoint.
-func NewAdminResolver(
+// NewServers initializes the GraphQL servers.  It sets up an empty server for the
+// main /graphql endpoint and an admin server.  The result is mainServer, adminServer.
+func NewServers(
+	config *ConnectionConfig, withIntrospection bool) (web.IServeGraphQL, web.IServeGraphQL) {
+
+	gqlSchema, err := schema.FromString("")
+	if err != nil {
+		panic(err)
+	}
+
+	rf := resolve.NewResolverFactory()
+	if withIntrospection {
+		rf.WithSchemaIntrospection()
+	}
```

Comment:
this is the only change.  The rest was just code move, because more is about to happen on startup and it belongs in the admin setup, not run.go